src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -745,6 +747,73 @@ class StringGenerator : public Generator<std::string> {
     const size_t _alphabetLength;
 };
 
+/** `{^ChooseFromDataset:{...}` */
+class RandomDataFromDataset : public Generator<std::string> {
+
+public:
+    RandomDataFromDataset(const Node& node, GeneratorArgs generatorArgs)
+        : _rng{generatorArgs.rng},
+          _id{generatorArgs.actorId},
+          _path{node["path"].maybe<std::string>().value()},
+          _pathLength{_path.size()}
+      {
+
+
+        if (_pathLength <= 0) {
+            BOOST_THROW_EXCEPTION(InvalidValueGeneratorSyntax(
+            "RandomDataFromDataset requieres non-empty path"));
+            }
+
+            /* To avoid reading and storing the datasets multiple times
+               we check if the path already exists in the map (as key). If
+               it doesn't exist, we open the file and store its contents. 
+               Section locked with a mutex */
+            
+            _dataset_mutex.lock();
+            if (_dataset.count(_path) == 0) {
+                std::cout << "I enter to open " << _path << std::endl;
```

Comment:
Is this a debug statement or one to leave in? If it's one to leave in, please use `BOOST_LOG_TRIVIAL`, example [here](https://github.com/mongodb/genny/blob/fb48f0ea6056d920ab6b3a93c5470a977c03c288/src/value_generators/src/DocumentGenerator.cpp#L991).

Reply:
(nit) also consider a non-first-person statement like `Loading dataset at <PATH>` or so. Also consider making it a debug-level statement.

Reply:
I forgot to remove that line, thank you :)